crates/meilisearch/tests/vector/mod.rs

```diff
@@ -161,7 +161,57 @@ async fn add_remove_user_provided() {
       "limit": 20,
       "total": 1
     }
-    "###);
+    "#);
+}
+
+#[actix_rt::test]
+async fn user_provide_mismatched_embedding_dimension() {
+    let server = Server::new().await;
+    let index = server.index("doggo");
+
+    let (response, code) = index
+        .update_settings(json!({
+          "embedders": {
+              "manual": {
+                  "source": "userProvided",
+                  "dimensions": 3,
+              }
+          },
+        }))
+        .await;
+    snapshot!(code, @"202 Accepted");
+    server.wait_task(response.uid()).await.succeeded();
+
+    let documents = json!([
+      {"id": 0, "name": "kefir", "_vectors": { "manual": [0, 0] }},
+    ]);
```

Comment:
```suggestion
    let documents = json!([
      {"id": 0, "name": "kefir", "_vectors": { "manual": [[0, 0], [1, 1], [2, 2] }},
    ]);
```

If I understand correctly in the FIXME you’re saying that this case will pass when it shouldn’t?
Can we test it?
Maybe it would be better to keep the beginning of your test as-is since it’s very easy to understand and at the end add a new test with an input that shows the issue

Reply:
Thank you for the rapid resonse! 🙌 Yes, this is indeed the case I was referring to in my comments. I will add another test at the end of the test function to illustrate the issue.

Would it be a good idea to address this case in this PR?

After reviewing the code and considering potential solutions, I realized that resolving this case requires a non-trivial change. My thought was to provide embedder information to the extractor like below. 

from :

https://github.com/meilisearch/meilisearch/blob/eefefc482b46dfa8368db13973455c7c92333dca/crates/milli/src/update/new/channel.rs#L511-L516 

``` rust
// to
 fn set_vectors( 
     &self, 
     docid: u32, 
     embedder_id: u8, 
     embedder : &Embedder
     embeddings: &[Vec<f32>], 
 ) -> crate::Result<()> { 
```

Given the complexity of the fix, I propose documenting the issue with a test in this PR and opening a separate PR later to properly address it. Since embedding dimensions are generally large nowadays, this should be a relatively low-priority fix.

Let me know what you think

Reply:
Yep, definitely you’re right.
That’s very ugly, but it shouldn’t happen often, and most users only have one vector anyway.

Just pinging @dureuill because they might be interested